packages/devtools_app/lib/src/screens/memory/allocation_profile_table_view.dart

```diff
@@ -0,0 +1,424 @@
+        if (profile == null) {
+          return const CenteredCircularProgressIndicator();
+        }
+        return ValueListenableBuilder<bool>(
+          valueListenable: preferences.vmDeveloperModeEnabled,
+          builder: (context, vmDeveloperModeEnabled, _) {
+            return ElevatedCard(
+              padding: const EdgeInsets.only(),
+              child: LayoutBuilder(
+                builder: (context, constraints) {
+                  return FlatTable<ClassHeapStats?>(
+                    columnGroups: [
+                      ..._columnGroups,
+                      if (vmDeveloperModeEnabled)
+                        ..._vmDeveloperModeColumnGroups,
+                    ],
+                    columns: [
+                      ..._columns,
+                      if (vmDeveloperModeEnabled) ..._vmDeveloperModeColumns,
+                    ],
+                    data: profile.members!.where(
+                      (element) {
+                        return element.bytesCurrent != 0 ||
+                            element.newSpace.externalSize != 0 ||
+                            element.oldSpace.externalSize != 0;
+                      },
+                    ).toList(),
+                    sortColumn: _columns.first,
+                    sortDirection: SortDirection.ascending,
+                    onItemSelected: (item) => null,
+                    keyFactory: (d) => Key(d!.classRef!.name!),
+                  );
+                },
+              ),
+            );
+          },
+        );
+      },
+    );
+  }
+}
+
+class _AllocationProfileTableControls extends StatelessWidget {
+  const _AllocationProfileTableControls({
+    Key? key,
+    required this.allocationProfileController,
+    this.refreshOnGcKey,
+    this.refreshKey,
+  }) : super(key: key);
+
+  final AllocationProfileTableViewController allocationProfileController;
+  final Key? refreshOnGcKey;
+  final Key? refreshKey;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        _ExportAllocationProfileButton(
+          allocationProfileController: allocationProfileController,
+        ),
+        const SizedBox(width: denseSpacing),
+        OutlinedIconButton(
+          key: refreshKey,
+          icon: Icons.refresh,
+          onPressed: () => allocationProfileController.refresh(),
+        ),
+        const SizedBox(width: denseSpacing),
+        _RefreshOnGCToggleButton(
+          allocationProfileController: allocationProfileController,
+          refreshOnGcKey: refreshOnGcKey,
+        ),
+      ],
+    );
+  }
+}
+
+class _ExportAllocationProfileButton extends StatelessWidget {
+  const _ExportAllocationProfileButton({
+    Key? key,
+    required this.allocationProfileController,
+  }) : super(key: key);
+
+  final AllocationProfileTableViewController allocationProfileController;
+
+  @override
+  Widget build(BuildContext context) {
+    return ExportButton(
```

Comment:
Let's customize the tooltip for the button: 'Download data in csv format', so that user knows where to search for the file and what format it will be.